### ISSUE TITLE
Inconsistent behavior of `# type: ignore` for whole file between python versions

### ISSUE THREAD

**Issue by isac322** [python/mypy#7785]
* **_Are you reporting a bug, or opening a feature request?_**

A bug

* **_Please insert below the code you are checking with mypy_**

`test.py`:
```python
# type: ignore

def test1():
    return 1

print(test1())
```

* **_What is the actual output?_**

with `mypy test.py --strict`:
```
Success: no issues found in 1 source file
```

with `mypy test.py --strict -2`:
```
test.py:1: error: unused 'type: ignore' comment
test.py:3: error: Function is missing a return type annotation
test.py:6: error: Call to untyped function "test1" in typed context
Found 3 errors in 1 file (checked 1 source file)
```

* **_What is the output you expect?_**

both `mypy test.py --strict` and `mypy test.py --strict -2`:
```
Success: no issues found in 1 source file
```

* **_What are the versions of mypy and Python you are using?_**

  * mypy: 0.740
  * python: 3.7.4

* **_Do you see the same issue after installing mypy from Git master?_**

Yes (on 09c1fc7)

**Comment by brandtbucher** [comment]
I see the issue here. I'll have a patch up soon.

Nice catch @isac322!
